src/public/utils.js

```diff
@@ -0,0 +1,185 @@
+	if (!date)
+		return "";
+	date = new Date(date)
+	return date.toLocaleString().split(/[\s,]+/)[0];
+}
+
+export function utils_fmt_history_img(val) {
+    const history_img_map = {
+        "pagamento_saldo": IMG_PAGAMENTO_SALDO,
+        "cashback": IMG_CASHBACK,
+        "pagamento_dinheiro": IMG_PAGAMENTO_DINHEIRO,
+        "pagamento_cartao": IMG_PAGAMENTO_CARTAO,
+    };
+    return history_img_map[val] ?? "";
+}
+
+export function utils_fmt_history_type(val) {
+    const history_type_map = {
+        "pagamento_saldo": TEXT_PAGAMENTO_SALDO,
+        "cashback": TEXT_CASHBACK,
+        "pagamento_dinheiro": TEXT_PAGAMENTO_DINHEIRO,
+        "pagamento_cartao": TEXT_PAGAMENTO_CARTAO,
+    };
+    return history_type_map[val] ?? "";
+}
+
+
+// -------------- config functions --------------------
+export function utils_config_items($w, data) {
+	let config = g_hist_map;
+	for (let item of config) {
+		let k = item.ui;
+		let val = item.db ? data[item.db] : item.raw;
+		val = item.fmt_from_db ? item.fmt_from_db(val) : val;
+		
+		let type = item.type;
+		
+		// add mask as format function
+		if (item.mask) {
+			item.format = (value) => {
+				return apply_mask(item.mask, value);
+			}
+		} else if ($w(k).onInput) {
+			$w(k).maxLength = 200;
+		}
+
+		switch (type) {
+			case "download":
+			case "checkbox":
+			case "upload":
+			case "button":
+				break;
+			case "text":
+				val = val || "";
+				// falls through
+			default:
+				if (!val && item.hide_on_empty && $w(k).hide)
+					$w(k).hide();
+				else {
+					var new_val = item.format ? item.format(val) : val;
+					$w(k)[type] = type == "text" && new_val.toString ? new_val.toString() : new_val;
+				}
+				break;
+		}
+
+		if (item.label)
+			$w(k).label = item.label;
+
+		if (item.onChange)
+			$w(k).onChange( (event) => { item.onChange(event, val) });
+		
+		if (item.onClick)
+			$w(k).onClick( (event) => { item.onClick(event, val) });
+
+		if (item.format && $w(k).onInput) {
+			$w(k).onInput( (event) => {
+				if (event.target.value)
+					$w(k)[type] = item.format(event.target.value);
+			});
+		}
+	}
+}
+
+
+// -------------- actions functions --------------------
+export function utils_show_hide_section(show_list, hide_list) {
+    for (let item of hide_list) {
+        $w(item).hide && $w(item).hide();
+        $w(item).collapse && $w(item).collapse();
+    }
+    for (let item of show_list) {
+        $w(item).expand && $w(item).expand();
+        $w(item).show && $w(item).show();
+    }
+}
+
+
+// function on 'utils' to be used by 'Extrato' and 'Home' pages
+export function utils_set_sections_history(state) {
+    switch (state) {
+        case SECTION_STATE_LOADING:
+            return utils_show_hide_section(["#sectionLoading"], ["#sectionHistData, #sectionNoData"]);
+        case SECTION_STATE_DATA:
+            return utils_show_hide_section(["#sectionHistData"], ["#sectionLoading, #sectionNoData"]);
+		case SECTION_STATE_NO_DATA:
```

Comment:
identation

Reply:
Wix bug I guess, on my editor is ok. I'll add to TODO board list to correct this and others style problems that may appear xD

Reply:
you can add a formatter on vscode. I added a formater that adjust the style when I save the file.